### PR TITLE
fix(checkbox): hide check icon from screen readers

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #5131 

## Summary

Updates check SVG to hide the check icon from screen readers. cc @geospatialem

### CODE REVIEW

File: src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -226,7 +226,7 @@ export class Checkbox implements LabelableComponent, CheckableFormComponent, Int
           role="checkbox"
           tabIndex={this.disabled ? undefined : 0}
         >
-          <svg class="check-svg" viewBox="0 0 16 16">
+          <svg aria-hidden="true" class="check-svg" role="none" viewBox="0 0 16 16">
```
Comment (geospatialem) [inline src/components/checkbox/checkbox.tsx:229]: We won't need the `role` attribute for Chrome/VoiceOver skip over the svg. So if we're looking to limit attributes, we can get this one down to one new attribute. ✂️ 

`<svg aria-hidden="true" class="check-svg" viewBox="0 0 16 16">`

**Review by geospatialem — COMMENTED** [review]
Klara was able to identify the culprit combination, which is VoiceOver and Chrome. All other screen readers are already avoiding the SVG. To support the case, we will only need the `aria-hidden` attribute. 👍🏻

**Comment by jcfranco** [comment]
@geospatialem Belated thanks for the review and recommendations. I've made the suggested changes and will proceed to install. Let me know if we need to tweak anything after. cc @benelan

**Comment by jcfranco** [comment]
Failing test is unrelated. Proceeding with merge.

**Pull request merged**